default.hbs

```diff
@@ -76,23 +73,123 @@
+                // Check if there is a need to filter posts
+                if (filters.length > 0) {
+                    // Hide all posts by default
+                    $('.gh-postfeed').children().each(function() {
+                        $(this).addClass('d-none');
+                    });
+                    // Show post if post meets the filter tags for each filter 
+                    $.each(filters, function(index, value) {
+                        $('.gh-postfeed').children().each(function() {
+                            if ($(this).hasClass(value)) {
+                                $(this).removeClass('d-none');
+                            }
+                        });
+                    });
+                } else {
+                    // No filter needed, show all posts
+                    $('.gh-postfeed').children().each(function() {
+                        $(this).removeClass('d-none');
+                    });
+                }
+            };
+        
+            function retrieveFiltersFromLS(){
+                // Retrieve filters from local storage and set it to the respective select element (force refresh using change())
+                let rawIntFilter = localStorage.getItem('internalStories').split("|");
+                $('#internal-filter').val(rawIntFilter).change();
+                let rawPriFilter = localStorage.getItem('primaryStories').split("|");
+                $('#primary-filter').val(rawPriFilter).change();
+            }
             $(document).ready(function () {
-                // Mobile Menu Trigger
+                // On page load, set persisted filters via localstorage 
+                retrieveFiltersFromLS();
+                // Filter posts on active page
+                filterPosts();
+                // Listen to changes to both filters
+                $('#primary-filter, #internal-filter').change(function(){
+                    // Retrieve filter options on selection change
+                    {{!-- getFilterOptions(this); --}}
+                    // Filter 
+                    filterPosts();
+                })
+                    // Mobile Menu Trigger
                 $('.gh-burger').click(function () {
                     $('body').toggleClass('gh-head-open');
                 });
+                // Filtering
+                $('.filter-selection').click(function(event) {
+                    console.log("Filter");
+                    let classToFilter = event.target.innerHTML;
+                    console.log("Filtering: " + classToFilter);
+                    let internalTag = $(this).hasClass('internal-tag');
+                    // Basic slug generator: replace whitespaces with dashes, then strip punctuation
+                    // TODO: Document behaviour of this generator in API v3
+                    classToFilter = classToFilter.replace(/\s+/g, '-').replace(/[.,'!?]/, '').toLowerCase();
```

Comment:
Could this be abstracted into a separate function, since it was used on L104 and L178 (here)? To clarify my comment on L177, the conversion into a slug is not well-documented in the Ghost docs. So, if we reuse this regex it should be placed into a separate function with a note that this may break at any time.

Reply:
Fixed as part of f1e4169600a1c7305e809dc571697d11841dec8a